test/integration/acceptance/add_signatory_test.cpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "integration/acceptance/acceptance_fixture.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+
+class AddSignatory : public AcceptanceFixture {
+ public:
+  auto makeFirstUser(const interface::RolePermissionSet &perms = {
+                         interface::permissions::Role::kAddSignatory}) {
+    return AcceptanceFixture::makeUserWithPerms(perms);
+  }
+
+  auto makeSecondUser(const interface::RolePermissionSet &perms = {
+                          interface::permissions::Role::kReceive}) {
+    return complete(
+        createUserWithPerms(kUser2, kUser2Keypair.publicKey(), kRole2, perms)
+            .creatorAccountId(kAdminId),
+        kAdminKeypair);
+  }
+
+  const std::string kRole2 = "roletwo";
+  const std::string kUser2 = "usertwo";
+  const std::string kUser2Id = kUser2 + "@test";
+  const std::string kAmount = "1.0";
+  const crypto::Keypair kUser2Keypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+};
+
+#define CHECK_BLOCK(i) \
```

Comment:
There was no consensus about the usage of macros, maybe discuss it with the team before merging

Reply:
It was merged earlier, so it just copy-pasted from some other test
I'd better refactor it as a static function of AcceptanceFixture in future